lnst/Tests/TrafficControl.py

```diff
@@ -1,45 +1,48 @@
-import logging
-import subprocess
+import shutil
+import asyncio
 import time
 
-from lnst.Common.Parameters import StrParam
+from lnst.Common.Parameters import StrParam, ListParam
 from lnst.Tests.BaseTestModule import BaseTestModule
 
 
 class TrafficControlRunner(BaseTestModule):
+    batchfiles = ListParam(type=StrParam(), required=True)
 
-    batchfile = StrParam(required=True)
-
-    def run(self) -> bool:
+    def run(self):
         self._res_data = {}
-
-        start_timestamp, time_taken, res = self.run_tc(self.params.batchfile)
-        success = res.returncode == 0
-
-        self._res_data["data"] = {
-            "time_taken": time_taken,
-            "start_timestamp": start_timestamp
-        }
-
-        msg = f"took {time_taken} sec"
-        if success:
-            self._res_data["msg"] = f"tc run successful, {msg}"
-            logging.info(self._res_data["msg"])
-        else:
-            self._res_data["msg"] = f"tc run failed, {msg}"
-            self._res_data["stderr"] = res.stderr
-            logging.warning(self._res_data["msg"])
-            logging.error(self._res_data["stderr"])
-
-        return success
-
-    def run_tc(self, batchfile: str) -> tuple[float, subprocess.CompletedProcess]:
-
+        instance_results = asyncio.run(self.run_instances())
+        all_success = all((i["success"] for i in instance_results))
+        self._res_data["data"] = dict(
+            instance_results=instance_results,
+        )
+        return all_success
+
+    async def run_instances(self):
+        tc_exec = shutil.which("tc")
+        instances = [
+            self.run_tc(tc_exec, bf)
+            for bf in self.params.batchfiles
+        ]
+        results = await asyncio.gather(*instances)
+        return results
+
+    async def run_tc(self, tc_exec: str, batchfile: str) -> dict[str]:
         start_timestamp = time.time()
-        st = time.perf_counter()
-        res = subprocess.run(f"tc -b {batchfile}", shell=True)
-        et = time.perf_counter()
-
-        time_taken = et - st
-
-        return start_timestamp, time_taken, res
+        start_time = time.perf_counter()
+        proc = await asyncio.create_subprocess_exec(
+            tc_exec, "-b", batchfile,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+        stdout, stderr = await proc.communicate()
```

Comment:
the previous version of the test module also logged any errors, this is probably a good idea to do in this version as well, especially since we're now running multiple processes at once - having more debug info could be useful in case something goes wrong.

Reply:
looks good now